src/conf/FaasmConfig.cpp

```diff
@@ -18,7 +18,7 @@ void FaasmConfig::initialise()
     // Threads
     moduleThreadPoolSize = this->getIntParam("FAASM_THREAD_POOL_SIZE", "0");
     if (moduleThreadPoolSize == 0) {
-        moduleThreadPoolSize = faabric::util::getUsableCores() + 2;
+        moduleThreadPoolSize = faabric::util::getUsableCores();
```

Comment:
Overloading really destroys performance. Note that this is the thread pool size for each module which itself will be created lazily.